packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -503,11 +503,13 @@ abstract class IosLLDBInit extends Target {
     // an error.
     final String? srcRoot = environment.defines[kSrcRoot];
     if (srcRoot == null) {
-      throw MissingDefineException(kSdkRoot, name);
+      environment.logger.printError('Failed to find $srcRoot');
+      return;
```

Comment:
Why change this one? It looks like the unpack ios target throws a missing define exception for this case too:

https://github.com/flutter/flutter/blob/cd433d4d365b25e880162ce006bda8eb838c6138/packages/flutter_tools/lib/src/build_system/targets/ios.dart#L265-L268

Reply:
Just to be safe. Basically, if this target fails, we don't want it to stop the build.

Reply:
Personally I would keep the exceptions for things that are unexpected and indicate a bug in our logic that we should investigate. I don't feel strongly though, feel free to keep as is.